application/application_client.go

```diff
@@ -0,0 +1,86 @@
+package application
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/hashicorp/go-oracle-terraform/client"
+	"github.com/hashicorp/go-oracle-terraform/opc"
+)
+
+// ApplicationClient represents an authenticated application client, with credentials and an api client.
+type ApplicationClient struct {
+	client *client.Client
+}
+
+func NewApplicationClient(c *opc.Config) (*ApplicationClient, error) {
+	appClient := &ApplicationClient{}
+	client, err := client.NewClient(c)
+	if err != nil {
+		return nil, err
+	}
+	appClient.client = client
+
+	return appClient, nil
+}
+
+func (c *ApplicationClient) executeCreateUpdateRequest(method, path string, files map[string]string, parameters map[string]interface{}) (*http.Response, error) {
+	req, err := c.client.BuildMultipartFormRequest(method, path, files, parameters)
+	if err != nil {
+		return nil, err
+	}
+
+	debugReqString := fmt.Sprintf("HTTP %s Path (%s)", method, path)
+	// req.Header.Set("Content-Type", "multipart/form-data")
+	// Log the request before the authentication header, so as not to leak credentials
+	c.client.DebugLogString(debugReqString)
+	c.client.DebugLogString(fmt.Sprintf("Req (%+v)", req))
+
+	// Set the authentication headers
+	req.SetBasicAuth(*c.client.UserName, *c.client.Password)
+	req.Header.Add("X-ID-TENANT-NAME", *c.client.IdentityDomain)
```

Comment:
This is fine as-is, but I was unclear in my previous review, you can do an unexported `const` like:

```golang
const tenantHeader = "X-ID-TENANT-NAME"
```